src/main/scala/viper/gobra/translator/encodings/slices/SliceEncoding.scala

```diff
@@ -144,6 +145,85 @@ class SliceEncoding(arrayEmb : SharedArrayEmbedding) extends LeafTypeEncoding {
     } yield withSrc(vpr.EqCmp(vLhs, vRhs), src)
   }
 
+  /**
+    * Encodes the allocation of a new slice
+    *
+    *  [r := make([]T, len, cap)] ->
+    *    var a []T@
+    *    inhales cap(a) == [cap]
+    *    inhales len(a) == [len]
+    *    inhales forall i: int :: {loc(a, i)} 0 <= i && i < [cap] ==> Footprint[ a[i] ]
+    *    inhales forall i: int :: {loc(a, i)} 0 <= i && i < [len] ==> [ a[i] == dfltVal(T) ]
+    */
+  override def statement(ctx: Context): in.Stmt ==> CodeWriter[vpr.Stmt] = {
+    default(super.statement(ctx)) {
+      case makeStmt@in.MakeSlice(target, typeParam, lenArg, optCapArg) =>
+        val (pos, info, errT) = makeStmt.vprMeta
+        val slice = in.LocalVar(Names.freshName, in.SliceT(typeParam.withAddressability(Shared), Addressability.Exclusive))(makeStmt.info)
+        val it = in.LocalVar(Names.freshName, in.IntT(Addressability.Exclusive))(makeStmt.info)
+        seqn(
+          for {
+            // var a []T@
+            _ <- local(ctx.typeEncoding.variable(ctx)(slice))
+
+            capArg = optCapArg.getOrElse(lenArg)
+            vprLength <- ctx.expr.translate(lenArg)(ctx)
+            vprCapacity <- ctx.expr.translate(capArg)(ctx)
+
+            // Perform additional runtime checks of conditions that must be true when make is invoked, otherwise the program panics (according to the go spec)
+            _ <- write(vpr.Inhale(vpr.LeCmp(vpr.IntLit(0)(pos, info, errT), vprLength)(pos, info, errT))(pos, info, errT)) // 0 <= len
+            _ <- write(vpr.Inhale(vpr.LeCmp(vpr.IntLit(0)(pos, info, errT), vprCapacity)(pos, info, errT))(pos, info, errT)) // 0 <= cap
+            _ <- write(vpr.Inhale(vpr.LeCmp(vprLength, vprCapacity)(pos, info, errT))(pos, info, errT)) // len <= cap
+            _ <- errorT {
+              case err.InhaleFailed(Source(info), _, _) => PreconditionError(info).dueTo(MakePreconditionError(info))
+            }
+
+            lenExpr = in.Length(slice)(makeStmt.info)
+            capExpr = in.Capacity(slice)(makeStmt.info)
+
+            // inhales cap(a) == [cap]
+            eqCap <- ctx.typeEncoding.equal(ctx)(capExpr, capArg, makeStmt)
+            _ <- write(vpr.Inhale(eqCap)(pos, info, errT))
+
+            // inhales len(a) == [len]
+            eqLen <- ctx.typeEncoding.equal(ctx)(lenExpr, lenArg, makeStmt)
+            _ <- write(vpr.Inhale(eqLen)(pos, info, errT))
+
+            // inhales forall i: int :: {loc(a, i)} 0 <= i && i < [cap] ==> Footprint[ a[i] ]
+            vprIt = ctx.typeEncoding.variable(ctx)(it)
+            footprint <- addressFootprint(ctx)(in.IndexedExp(slice, it)(makeStmt.info), in.FullPerm(slice.info))
```

Comment:
Gobra runs into an Exception here. For some reason, the condition `typ(ctx).isDefinedAt(t)` in the body of the `addressFootprint` method is not true, which causes Gobra to crash. @Felalolf, do you have any idea of why this happens?